### PR TITLE
Job duration time is correct now. Console output endpoint was removed

### PR DESCRIPTION
**Pull request by ubuntolog**
Job duration time sometimes was a negative number. Now we calculate the duration time on the server side to avoid conflicts with time zones and incorrect local time settings. Console output endpoint was not used any longer and therefore was removed.

### CODE REVIEW

File: gefserver/db/db.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ type JobTable struct {
 	ServiceID    string
 	Input        string
 	Created      time.Time
-	Finished     time.Time
+	Duration     int64
```
Comment (emanueldima) [inline gefserver/db/db.go]: add comment // duration time in seconds

File: webui/src/components/Jobs.jsx [hunk 2]
```diff
@@ -147,13 +161,12 @@ class Jobs extends React.Component {
                 (service && service.ID && service.ID.length) ? service.ID : "unknown service";
             let title = "Job from " + serviceName;
 
-            let execDuration = "";
-            if (job.State.Code == -1) {
-                let currentDate = new Date();
-                execDuration = currentDate - Date.parse(job.Created);
+            let jobStartTime = new Date(job.Created);
+            let jobDurationTime = new Date(job.Duration);
+            let jobFinishTime = new Date(jobStartTime.getTime() + jobDurationTime.getTime());
```
Comment (emanueldima) [inline webui/src/components/Jobs.jsx]: This seems incorrect, server duration is in seconds and new Date accepts milliseconds. You should use:
`let jobFinishTime = new Date(jobStartTime.getTime() + 1000 * job.Duration);`

File: gefserver/tests/server_test.go [hunk 3]
```diff
@@ -158,16 +157,6 @@ func TestServer(t *testing.T) {
 	}
 	jobOutputVolume := job["OutputVolume"].(string)
```
Comment (emanueldima) [inline gefserver/tests/server_test.go]: add a test here for the console output

File: gefserver/tests/server_test.go [hunk 4]
```diff
@@ -157,6 +161,27 @@ func TestServer(t *testing.T) {
...
+			taskMap := s.Index(1).Interface().(map[string]interface{})
+
+			for key, value := range taskMap {
+				if key == "ConsoleOutput" {
+					console = value
+					break
+				}
+			}
+		}
+	}
```
Comment (emanueldima) [inline gefserver/tests/server_test.go]: why not:
````
tasks := job['Tasks'].([]interface{})
if len(tasks)>1 {
    console = tasks[1].(map[string]interface{})['ConsoleOutput']
}
````

Comment (ubuntolog) [inline gefserver/tests/server_test.go]: It looks much cleaner. I got stuck a bit with interface{} conversions.

**Pull request merged**